docs/Marketplace.md

```diff
@@ -0,0 +1,360 @@
+```json
+{"id":"16Uiu2HAm4yoghEaNy1VPLumvXwrFSEenzG1zSy3EcTFYLE2mynjp","addrs":["/ip4/0.0.0.0/tcp/8070"],"repo":"/data","spr":"spr:CiUIAhIhAo30ewLoBnzU5COMicomRT6UAfdmDJbdwQRnba0cvHnnEgIDARo8CicAJQgCEiECjfR7AugGfNTkI4yJyiZFPpQB92YMlt3BBGdtrRy8eecQqrWcsAYaCwoJBLNdyciRAh-aKkYwRAIgFnH5tSmOlxrSnOwyOHjnjBVSrEXZOeosH1i9gzoDI6UCIEnCI82Bb4iATMKKuaZCfVNlEOmwnfAG-6z0RuJJc-8s",...
+```
+The part we care about is the SPR (the string after `"spr":`). Write that string down.
+
+**Starting the client node.** The client node is started similarly except that:
+
+* we need to pass the SPR of the storage node so it can form a network with it;
+* since it does not run any proofs, it does not require any ceremony files.
+
+```bash
+${CODEX_BINARY}\
+  --data-dir=./codex-client\
+  --listen-addrs=/ip4/0.0.0.0/tcp/8081\
+  --api-port=8001\
+  --disc-port=8091\
+  --bootstrap-node=<storage-node-spr>
+  persistence\
+  --eth-provider=http://localhost:8545\
+  --eth-private-key=<(echo -n "0x5538ec03c956cb9d0bee02a25b600b0225f1347da4071d0fd70c521fdc63c2fc")\
+  --marketplace-address=${MARKETPLACE_ADDRESS}
+```
+
+## 4. Buy and Sell Storage on the Marketplace
+
+Any storage negotiation has two sides: a buyer and a seller. Before we can actually request storage, therefore, we must first put some of it for sale.
+
+### 4.1 Sell Storage
+
+The following request will cause the storage node to put out $50\text{MB}$ of storage for sale for $1$ hour, at a price of $1$ Codex token per byte per second, while expressing that it's willing to take at most a $1000$ Codex token penalty for missing a storage proof.
+
+```bash
+curl 'http://localhost:8000/api/codex/v1/sales/availability' \
+  --header 'Content-Type: application/json' \
+  --data '{
+  "totalSize": "50000000",
+  "duration": "3600",
+  "minPrice": "1",
+  "maxCollateral": "1000"
+}'
+```
+
+this should return a response with an id a string (e.g. `"id": "0x552ef12a2ee64ca22b237335c7e1df884df36d22bfd6506b356936bc718565d4"`) which identifies this storage offer. To check the current storage offers for this node, you can issue:
+
+```bash
+curl 'http://localhost:8000/api/codex/v1/sales/availability'
+```
+
+this should print a list of offers, with the one you just created figuring among them.
+
+## 4.2. Buy Storage
+
+Before we can buy storage, we must have some actual data to request storage for. Start by uploading a small file to your client node. On Linux you could, for instance, use `dd` to generate a $100KB$ file:
+
+```bash
+dd if=/dev/urandom of=./data.bin bs=100K count=1
+```
+
+but any small file will do. Assuming your file is named `data.bin`, you can upload it with:
+
+```bash
+curl "http://localhost:8001/api/codex/v1/data" --data-bin @data1.bin
+```
+
+Once the upload completes, you should see a CID (e.g. `zDvZRwzm2mK7tvDzKScRLapqGdgNTLyyEBvx1TQY37J2CdWdS6Sj`) for the file printed to the terminal. Use that CID in the purchase request:
+
+```bash
+curl "http://localhost:8081/api/codex/v1/storage/request/<upload CID>"
```

Comment:
```suggestion
curl "http://localhost:8001/api/codex/v1/storage/request/<upload CID>"
```

Reply:
I'm getting an error I think you ran into previously: `Unable to encode manifest, not enough blocks, ecK = 2, blocksCount = 1%`